pkg/query/endpointset.go

```diff
@@ -203,6 +210,9 @@ func (c *endpointSetNodeCollector) Update(nodes map[component.Component]map[stri
 	for k, v := range nodes {
 		storeNodes[k] = make(map[string]int, len(v))
 		for kk, vv := range v {
+			if len(kk) > externalLabelLimit {
+				kk = kk[:externalLabelLimit+1] + "}"
```

Comment:
I wonder if it makes sense to pull this to small function and really test it properly on boundaries.

Reply:
Pulling into small function makes sense but is testing the boundaries really important? Since the main goal is not to have super long label names, does it matter if it's trimmed to 1001 or 1000? Especially since it will be consistent. There are curly braces `{` around the external label string, so the +1 kinda excludes it in the braces

Reply:
Well, what if we go into IF and add `}` for something that is not trimmed?

>  There are curly braces { around the external label string, so the +1 kinda excludes it in the length

In theory, but can we test it?

Reply:
> Since the main goal is not to have super long label names, does it matter if it's trimmed to 1001 or 1000? 

it does not - but it matters we you trim it to `{labels...}` or `{labels..}}` by accident. (:

Reply:
.... and there is indeed bug. Are you still sure more testing was not needed? (: 

https://github.com/thanos-io/thanos/pull/5785#discussion_r1000481280